inertia.go

```diff
@@ -84,6 +84,31 @@ func NewFromBytes(rootTemplateBs []byte, opts ...Option) (*Inertia, error) {
 	return New(string(rootTemplateBs), opts...)
 }
 
+// NewFromTemplate receives a *template.Template and then initializes Inertia.
+func NewFromTemplate(rootTemplate *template.Template, opts ...Option) (*Inertia, error) {
```

Comment:
I just think maybe now we should refuse of lazy template parsing?
I mean, we could remove the buildRootTemplate from the doHTMLResponse and move it to the `New` constructor?
In turn, the `New` constructor only parse the template from string and pass it to our new function `NewFromTemplate`.
This will eliminate duplication of initialization logic in constructors and it will immediately return an error if something is wrong in the template.

Reply:
We can also remove `rootTemplateHTML` from an `Inertia` struct

Reply:
Apparently, there's no built-in way to update the FuncMap of an instance of `template.Template`. So, `rootTemplateHTML` would be needed to support the following method and re-parse `rootTemplate` with the revised `sharedTemplateFuncs`:

```go

// ShareTemplateFunc adds passed value to the shared template func map.
func (i *Inertia) ShareTemplateFunc(key string, val any) error {
	i.sharedTemplateFuncs[key] = val
	
	// rebuilding root template with new template funcs
	if i.rootTemplateHTML == "" {
		return fmt.Errorf("no root template string defined")
	}

	t, err := template.New("root").
		Funcs(template.FuncMap(i.sharedTemplateFuncs)).
		Parse(i.rootTemplateHTML)
	if err != nil {
		return fmt.Errorf("rebuild root template: %w", err)
	}

	i.rootTemplate = t
	return nil
}

// original method
//func (i *Inertia) ShareTemplateFunc(key string, val any) {
//	i.sharedTemplateFuncs[key] = val
//}
```

Of course, my proposed `NewFromTemplate` constructor does not require `rootTemplateHTML` to be defined, so for that use case it would have to be assumed that any management of template.Funcs would have to be done externally. Thoughts?

Reply:
Yup, you are right, we can't do that.
I will look at your pull request in the near future and we will merge it.

Reply:
For my use case, users will need to ensure that `Template.Funcs` are defined before using the `NewFromTemplate` constructor. Perhaps we should alert them by returning an error if `ShareTemplateFunc` is called when no root template HTML is present?

```go
// ShareTemplateFunc adds the passed value to the shared template func map. If
// no root template HTML string has been defined, it returns an error.
func (i *Inertia) ShareTemplateFunc(key string, val any) error {
	if i.rootTemplateHTML == "" {
		return fmt.Errorf("undefined root template html string")
	}

	i.sharedTemplateFuncs[key] = val
	return nil
}
```

Reply:
I think it's right and we can merge it now, but maybe in the future we should abandon the shateTemplateFunc function and accept template functions in the constructor.